lib/new_address_geocoder/base.rb

```diff
@@ -0,0 +1,29 @@
+module AddressGeocoder
+  class Base
+
+    def initialize()
+    end
+
+    def suggested_addresses
+      # call lookup
+      # return response to address geocoder
+    end
+
+    def valid_address?
+      # call lookup
+      # return response to address geocoder
+    end
+
+    private
+
+    def lookup
+      require "urls/#{api}"
```

Comment:
I agree only require api when it is needed. it should be faster in this gem usage case